wagtail/users/views/users.py

```diff
@@ -180,57 +182,72 @@ def create(request):
     )
 
 
-@permission_required(change_user_perm)
-def edit(request, user_id):
-    user = get_object_or_404(User, pk=user_id)
-    can_delete = user_can_delete_user(request.user, user)
-    editing_self = request.user == user
+class Edit(EditView):
+
+    model = User
+    permission_policy = ModelPermissionPolicy(User)
+    form_class = get_user_edit_form()
+    template_name = "wagtailusers/users/edit.html"
+    index_url_name = "wagtailusers_users:index"
+    success_url = "wagtailusers_users:index"
+    edit_url_name = "wagtailusers_users:edit"
+    delete_url_name = "wagtailusers_users:delete"
+    success_message = _("User '{0}' updated.")
+    context_object_name = "user"
+
+    def get_page_title(self):
+        return "Editing {}".format(self.object.get_username)
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.object = self.get_object()
+        self.can_delete = user_can_delete_user(request.user, self.object)
+        self.editing_self = request.user == self.object
+
+    def save_instance(self):
+        instance = super().save_instance()
+        if self.object == self.request.user and "password1" in self.form.changed_data:
+            # User is changing their own password; need to update their session hash
+            update_session_auth_hash(self.request, self.object)
+        return instance
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({"editing_self": self.editing_self})
+        return kwargs
+
+    def run_before_hook(self):
+        for fn in hooks.get_hooks("before_edit_user"):
+            result = fn(self.request, self.object)
+            if hasattr(result, "status_code"):
+                return result
 
-    for fn in hooks.get_hooks("before_edit_user"):
-        result = fn(request, user)
-        if hasattr(result, "status_code"):
-            return result
-    if request.method == "POST":
-        form = get_user_edit_form()(
-            request.POST, request.FILES, instance=user, editing_self=editing_self
+    def run_after_hook(self):
+        for fn in hooks.get_hooks("after_edit_user"):
+            result = fn(self.request, self.object)
+            if hasattr(result, "status_code"):
+                return result
+
+    def get_success_buttons(self):
+        return (
+            [
+                messages.button(
+                    reverse("wagtailusers_users:edit", args=(self.object.pk,)),
```

Comment:
Avoid hard-coding the url names here, instead use `self.edit_url_name`

```suggestion
                    reverse(self.edit_url_name, args=(self.object.pk,)),
```